src/core/socket.c

```diff
@@ -2244,17 +2246,35 @@ static void socket_enter_start_chown(Socket *s) {
                 r = socket_chown(s, &s->control_pid);
                 if (r < 0) {
                         log_unit_warning_errno(UNIT(s), r, "Failed to spawn 'start-chown' task: %m");
-                        goto fail;
+                        socket_enter_stop_pre(s, SOCKET_FAILURE_RESOURCES);
+                        return;
                 }
 
                 socket_set_state(s, SOCKET_START_CHOWN);
         } else
                 socket_enter_start_post(s);
+}
 
-        return;
+static void socket_enter_start_open(Socket *s) {
+        int r;
 
-fail:
-        socket_enter_stop_pre(s, SOCKET_FAILURE_RESOURCES);
+        assert(s);
```

Comment:
I'd like to insert an assertion about socket state, something like the following (though not checked carefully):
```c
assert(IN_SET(s->state, SOCKET_DEAD, SOCKET_FAILED, SOCKET_START_PRE));
```

Reply:
Hmm, dunno. I added assert in `socket_enter_start_chown` because it depends on `_OPEN`. But here exists no strict dep/relationship between these states.

Reply:
Not checked in detail, but is it safe to call it accidentally when in other states, e.g. by already started socket?